src/components/static-map.js

```diff
@@ -197,14 +206,56 @@ export default class StaticMap extends PureComponent<StaticMapProps, State> {
     }
   }
 
+  _tryToApplyLatestMapChanges() {
```

Comment:
This looks subtle. If this was "copy-pasted", add link to the source, if not add comments to guide maintainers?

Reply:
this was copied from an internal project. once this is in react-map-gl, we'll replace our internal implementation with this one.

Reply:
We made an explicit effort to move away from custom diffing in v3. What property in your style is causing the failure, shouldn't we open an issue on mapbox-gl?

Reply:
is-stroke - is failing to diff
args: [
0: "layerId"
1: "is-stroke"
2: true
]
command: "setLayerProperty"